### PR TITLE
Rename earth_model to detector_model

### PR DESCRIPTION
**Pull request by austinschneider**
Just realized we forgot to rename these variables in the previous PR

### CODE REVIEW

**Comment by austinschneider** [comment]
Looks like there are a few other places that "earth" pops in that we might want to rename.

**Comment by austinschneider** [comment]
There is also a file in math called Coordinates.cxx that we can delete.

File: projects/injection/private/pybindings/LeptonInjector.cxx
Comment (austinschneider) [reply projects/injection/private/pybindings/LeptonInjector.cxx:1]: Yeah we can get rid of them. I think it's spread across at least two files

**Comment by nickkamp1** [comment]
> Looks like there are a few other places that "earth" pops in that we might want to rename.

Yeah, for example functions like `GetDetCoordPosFromEarthCoordPos` should probably be renamed, but it might make sense to do that on the coordinate transformation branch?

**Comment by austinschneider** [comment]
Finally, something is causing the build to fail pretty early on in the GitHub runners. We should fix that too.

**Comment by nickkamp1** [comment]
> There is also a file in math called Coordinates.cxx that we can delete.

I agree it seems to be legacy. But I think we should delete it in the coordinate transformation branch?

**Comment by nickkamp1** [comment]
> > Looks like there are a few other places that "earth" pops in that we might want to rename.
> 
> Yeah, for example functions like `GetDetCoordPosFromEarthCoordPos` should probably be renamed, but it might make sense to do that on the coordinate transformation branch?

Ok I actually went back and renamed every earth -> detector instance that doesn't interface with DetectorModel::GetEarthCoordPosFromDetCoordPos, since I think we should rename that function based on the updates in the coordinate transformation branch

**Comment by nickkamp1** [comment]
> Finally, something is causing the build to fail pretty early on in the GitHub runners. We should fix that too.

I think this is the last remaining issue on this branch before we can merge

**Comment by austinschneider** [comment]
Cool, yeah I'll make those changes in the coordinate transformation branch 👍

**Comment by austinschneider** [comment]
These latest cmake changes allowed one of the linux builds to work. We'll see if this extends to the rest of the builds invoked by cibuildwheel 🤞

**Comment by austinschneider** [comment]
The linux builds seem to be going just fine now, but the macos builds are still stuck on the delocate stage. It seems like it is complaining that it cannot find python. If it is trying to package the python shared object library into the wheel, then we could stop it from doing so with `--ignore-missing-dependencies`, but it may actually need this. In any case, it is strange that we're getting this error. It is complaining that `/Library/Frameworks/Python.framework/Versions/3.8/Python` is not found, but this file does exist and is the correct python library.

File: cmake/Packages/Python.cmake
Comment (nickkamp1) [inline cmake/Packages/Python.cmake:1]: Is there a reason we can use the Development meta component on mac? [The github issue Philip found](https://github.com/pypa/cibuildwheel/issues/639) and the included [link to the pyind11 docs](https://pybind11.readthedocs.io/en/stable/compiling.html#findpython-mode) seems to suggest the Development component won't be resolved by FindPython if libpythonXX.so doesn't exist, which I think is the case for the MacOS runners?

Comment (austinschneider) [reply cmake/Packages/Python.cmake:1]: I think I chose to do it based on some testing. But I do not remember if that was on my computer or a runner. It's worth trying the other way though.

Comment (nickkamp1) [reply cmake/Packages/Python.cmake:1]: I tried it on my own computer and got the same build error you got in [the most recent commit](https://github.com/Harvard-Neutrino/LeptonInjector/pull/43/commits/3af743d32e8951078b8bc339e88e9419deb44a90), related to pybinding methods in DarkNews[CrossSection,Decay].cxx

Comment (austinschneider) [reply cmake/Packages/Python.cmake:1]: This might be related to the error we're seeing when using Development.Module on macos: https://github.com/pybind/pybind11/issues/382
I'm trying again after adding the pybind11::python_link_helper to the interactions target, which will add these linking options on macos.

**Comment by austinschneider** [comment]
Okay, we're still getting the same error on [cd2f3a4](https://github.com/Harvard-Neutrino/LeptonInjector/pull/43/commits/cd2f3a4761a44a0c3ef1ad0045da9bea592e5465). I forgot that the Development.Module was conditional on cibuild wheel, so trying the combination of Development.Module and pybind11::python_link_helper on my local machine actually has the same linking error with python.

**Comment by austinschneider** [comment]
At least one of the macos build is finishing now!

**Comment by nickkamp1** [comment]
> At least one of the macos builds is finishing now!

awesome thanks Austin!!

**Comment by austinschneider** [comment]
@nickkamp1 we can go ahead and merge this once you have a chance to look things over!

**Review by nickkamp1 — APPROVED** [review]
Looks good, merge approved! Thanks for getting the ci build working Austin

**Pull request merged**
